recce/__init__.py

```diff
@@ -35,6 +35,11 @@ def is_ci_env():
     return False
 
 
+def is_recce_cloud_instance():
+    """Check if running in Recce Cloud instance."""
+    return os.environ.get("is_recce_cloud_instance", "false").lower() == "true"
```

Comment:
Could we change the env name to `RECCE_CLOUD_INSTANCE=true`?
The naming convention will be similar to other platforms. 
Ex: GITHUB_ACTION=true, CI=true, etc...